xonsh/base_shell.py

```diff
@@ -503,12 +503,14 @@ def settitle(self):
         """Sets terminal title."""
         env = builtins.__xonsh__.env  # pylint: disable=no-member
         term = env.get("TERM", None)
-        # Shells running in emacs sets TERM to "dumb" or "eterm-color".
-        # Do not set title for these to avoid garbled prompt.
+        # Shells running in emacs sets TERM to "dumb", "eterm-color"
+        # or xterm-256color (for OS X). Do not set title for these
+        # to avoid garbled prompt.
         if (term is None and not ON_WINDOWS) or term in [
             "dumb",
             "eterm-color",
             "linux",
+            "xterm-256color",
```

Comment:
I don't think that the title shouldn't be set if `$TERM` is `xterm-256color`, it's very commonly used.

Reply:
thanks for the heads up! out of curiosity, do you know what `Shells running in emacs sets TERM to ...` means above? Is that EMACS_MODE? maybe I can check for that instead?

(also a really naive question: what _is_ the title? why do we have to evaluate these seemingly unused fields with every prompt?)

Reply:
I'm not sure but that might be some kind of sub-shell in emacs. (I'm a vim user, the `:terminal` in my vim does not break when using title setting escape sequences.)

This *title* (set via `$TITLE`) is the terminal window title, which can be set by ANSI escape sequence - if the terminal emulator supports it. Most xterm compatible terminal emulators does (those that identify themselves as `xterm-256color`should I think), this escape sequence is usually part of the $PROMPT, but currently xonsh - due to `prompt_toolkit` - doesn't support setting it this way. I just submitted a PR to support it (and other ANSI OSC escape sequences): #3784

Reply:
Some other examples: https://github.com/xonsh/xonsh/pull/3781#issuecomment-695809527

Reply:
thanks so much for explaining! i'm going to revert this part then